src/client/hud.cpp

```diff
@@ -283,11 +283,25 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 {
 	u32 text_height = g_fontengine->getTextHeight();
 	irr::gui::IGUIFont* font = g_fontengine->getFont();
+
+	// Reorder elements by z_index
+	std::vector<size_t> ids;
```

Comment:
Rather than HUD indices you might rather use `HudElement *`. The validity of those is already checked in the first loop.

Reply:
I did that first but unfortunately the index is displayed later in case of error:

https://github.com/minetest/minetest/blob/8dd96eef296048b49de69140052536bf350404a9/src/client/hud.cpp#L382

So I went back to indices.

Reply:
Maybe store an `std::pair` containing both the index and the element itself? Haven't checked the code, but it could probably be worth the extra memory if it saves a couple of lookups.

Reply:
I'm not convinced it worth it. On a cpu point of view, the lookups cost is quite small regarding the rest of the drawing code.
A better optimization would be to sort only when adding/removing HUD elements. But that would be a bigger change I guess (we still need to know element id and to be able to access elements by id). Again, not sure it worth it.

Reply:
Ok, no worries :)